### PR TITLE
Tool flexibility

### PR DESCRIPTION
**Pull request by kevinl424**
- ability to initialize `ChatAgent` with subset of default tools with new parameter
- `add_tools` now allows users to add their own custom tools to be used by `ChatAgent`
- `remove_tool` allows users to remove both custom and default tools 
- README Usage section additions detailing how to use these new methods

### CODE REVIEW

File: src/memary/agent/base_agent.py [hunk 1]
```diff
@@ -435,19 +421,67 @@ def get_entity(self, retrieve) -> list[str]:
...
-                tools.append(FunctionTool.from_defaults(fn=self.locate))
-            elif tool == "Vision":
-                tools.append(FunctionTool.from_defaults(fn=self.vision))
-            elif tool == "Stocks":
-                tools.append(FunctionTool.from_defaults(fn=self.stock_price))
-            # elif tool == "News":
-            #     tools.append(FunctionTool.from_defaults(fn=self.get_news))
-
-        self.routing_agent = ReActAgent.from_tools(tools, llm=self.llm, verbose=True)
+    def _init_ReAct_agent(self):
```
Comment (seyeong-han) [inline src/memary/agent/base_agent.py:424]: This is huge!!

File: streamlit_app/app.py [hunk 2]
```diff
@@ -136,13 +136,12 @@ def get_models(llm_models, vision_models):
 
     tools = st.multiselect(
         "Select tools to include:",
-        # ["Search", "Location", "Vision", "Stocks", "News"], #all options available
-        # ["Search", "Location", "Vision", "Stocks", "News"],) #options that are selected by default
-        ["Search", "Location", "Vision", "Stocks"],  # all options available
-        ["Search", "Location", "Vision", "Stocks"],
+        ["search", "locate", "vision", "stocks"],  # all options available
+        ["search", "locate", "vision", "stocks"],
```
Comment (seyeong-han) [inline streamlit_app/app.py:140]: Could you move `# options that are selected by default` annotation next to the second list?

File: src/memary/agent/chat_agent.py [hunk 3]
```diff
@@ -43,8 +58,8 @@ def get_chat(self):
     def clearMemory(self):
         self.memory_stream.clear_memory()
         self.entity_knowledge_store.clear_memory()
-        
-       # print("removed from mem stream and entity knowdlege store ")
+
+        # print("removed from mem stream and entity knowdlege store ")
```
Comment (seyeong-han) [inline src/memary/agent/chat_agent.py:62]: What about showing the message with a log?
```
logging.info("removed from mem stream and entity knowdlege store ")
```

**Review by seyeong-han — COMMENTED** [review]
This is a huge progress!

**Review by seyeong-han — APPROVED** [review]
Loved to see more logging.
Thanks!

File: src/memary/agent/base_agent.py [hunk 4]
```diff
@@ -116,18 +111,9 @@ def __init__(
...
-        # news_tool = FunctionTool.from_defaults(fn=self.get_news)
-
         self.debug = debug
-        self.routing_agent = ReActAgent.from_tools(
-            [search_tool, locate_tool, vision_tool, stock_tool],
-            llm=self.llm,
-            verbose=True,
-        )
+        self.tools = {}
+        self._init_default_tools(default_tools=include_from_defaults)
```
Comment (kingjulio8238) [inline src/memary/agent/base_agent.py:116]: 🔥🔥

**Review by kingjulio8238 — APPROVED** [review]

**Pull request merged**